packages/component-library-react/src/HTMLHeading.test.tsx

```diff
@@ -0,0 +1,78 @@
+import { render, screen } from '@testing-library/react';
+import { createRef } from 'react';
+import { HTMLHeading } from './HTMLHeading';
+import '@testing-library/jest-dom';
+
+describe('HTMLHeading', () => {
+  it('renders a heading role element', () => {
+    render(<HTMLHeading level={1}>Breaking news</HTMLHeading>);
+
+    const heading = screen.getByRole('heading', {
+      name: 'Breaking news',
+    });
+
+    expect(heading).toBeInTheDocument();
+    expect(heading).toBeVisible();
+  });
+
+  it('renders a heading at heading level 1', () => {
+    render(<HTMLHeading level={1}>Breaking news</HTMLHeading>);
+
+    const heading = screen.getByRole('heading', {
+      name: 'Breaking news',
+      level: 1,
+    });
+
+    expect(heading).toBeInTheDocument();
+  });
+
+  it('renders an HTML h1 element', () => {
+    const { container } = render(<HTMLHeading level={1} />);
+
+    const heading = container.querySelector('h1:only-child');
+
+    expect(heading).toBeInTheDocument();
+  });
+
+  it('renders rich text content', () => {
+    render(
+      <HTMLHeading level={1}>
+        <strong>Breaking</strong> news
+      </HTMLHeading>,
+    );
+
+    const heading = screen.getByRole('heading', {
+      name: 'Breaking news',
+    });
+
+    const richText = heading.querySelector('strong');
+
+    expect(richText).toBeInTheDocument();
+  });
+
+  it('can be hidden', () => {
+    const { container } = render(<HTMLHeading level={1} hidden />);
+
+    const heading = container.querySelector(':only-child');
+
+    expect(heading).not.toBeVisible();
+  });
+
+  it('can have a custom class name', () => {
+    const { container } = render(<HTMLHeading level={1} className="large" />);
+
+    const heading = container.querySelector(':only-child');
+
+    expect(heading).toHaveClass('large');
+  });
+
+  it('supports ForwardRef in React', () => {
+    const ref = createRef<HTMLHeadingElement>();
+
+    const { container } = render(<HTMLHeading level={1} ref={ref} />);
+
+    const heading = container.querySelector(':only-child');
+
+    expect(ref.current).toBe(heading);
+  });
+});
```

Comment:
Deze test is niet volledig: h2-h6 worden helemaal niet getest, en `level: 7` ook niet.